cedar-policy-core/src/entities.rs

```diff
@@ -213,7 +213,7 @@ impl Entities {
                 schema
                     .action_entities()
                     .into_iter()
-                    .map(|e| (e.uid().clone(), Arc::unwrap_or_clone(e))),
```

Comment:
It took me a second to puzzle out why this actually gives a speed up, so I'll leave my reasoning here. Maybe worth elaborating in a comment.

We remove this `unwrap_or_clone`, but add a `make_mut` (which I assume must clone if there's another reference to the data) call inside TC computation, so at first glance it feels like that should be a wash. But, the TC was already computed for theentity map (which we just constructed, so one ref to each `Arc`), and we separately computed TC for the actions inside the schema constructor (while it had unique ownership). Put together these should mean we so never trigger a clone on the `make_mut` during TC compunction.